org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/move/MoveClass.scala

```diff
@@ -131,6 +131,7 @@ class MoveClass extends RefactoringExecutorWithWizard {
             (change :: textChanges, newFiles)
           case (change: NewFileChange, (textChanges, newFilesChanges)) =>
             (textChanges, change :: newFilesChanges)
+          case (other, _) => throw new AssertionError(s"Unexpected change $other")
```

Comment:
I don't see any difference between throwing `MatchError` and `AssertionError`. Actually, as @sschaef noticed, `MatchError` gives you more information. I understand the warning is gone, but then let's report the error in a better way. (or use `@unchecked` to silence the warning).